x-pack/plugins/upgrade_assistant/server/lib/es_deprecation_logging_apis.ts

```diff
@@ -30,17 +28,38 @@ export async function setDeprecationLogging(
 ): Promise<DeprecationLoggingStatus> {
   const { body: response } = await dataClient.asCurrentUser.cluster.putSettings({
     body: {
+      persistent: {
+        'logger.deprecation': isEnabled ? 'WARN' : 'ERROR',
+        'cluster.deprecation_indexing.enabled': isEnabled,
+      },
+      /*
+       * If we only set the persistent setting, we can end up in a situation in which a user has
+       * set transient on/off. And when toggling and reloading the page the transient setting will
+       * have priority over it thus "overriding" whatever the user selected.
+       */
       transient: {
         'logger.deprecation': isEnabled ? 'WARN' : 'ERROR',
+        'cluster.deprecation_indexing.enabled': isEnabled,
       },
     },
   });
 
   return {
-    isEnabled: isDeprecationLoggingEnabled(response),
+    isEnabled,
+    isLoggerDeprecationEnabled: isDeprecationLoggingEnabled(response),
   };
 }
 
+export function isClusterDeprecationLoggingEnabled(settings: any) {
+  const clusterDeprecationLoggingEnabled = ['default', 'persistent', 'transient'].reduce(
+    (currentLogLevel, settingsTier) =>
+      get(settings, [settingsTier, 'cluster', 'deprecation_indexing', 'enabled'], currentLogLevel),
```

Comment:
Minor nit: Here and in the docs we refer to settings using the dot-concatenated path, e.g. `'cluster.deprecation_indexing.enabled'`. And typically, when skimming code and seeing an array we're used to identifying it as a set of elements, not as a path. So it might make this code a little easier to read to write the path using the dot-concatenated format. Likewise for line 66.

```js
get(settings, `${settingsTier}.cluster.deprecation_indexing.enabled`, currentLogLevel),
```